src/Frontend/Modules/Search/Actions/Index.php

```diff
@@ -210,8 +210,8 @@ private function buildForm(): void
         $this->form = new FrontendForm('search', null, 'get', null, false);
 
         // could also have been submitted by our widget
-        if (!\SpoonFilter::getGetValue('q', null, '')) {
-            $_GET['q'] = \SpoonFilter::getGetValue('q_widget', null, '');
+        if (!$this->getRequest()->has('q')) {
+            $_GET['q'] = $this->getRequest()->query->get('q_widget');
```

Comment:
Not sure if I understand what you mean. $_GET doesn't get altered before this line.